File: entities/history.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type VerificationHistory struct {
	ID          uuid.UUID  `gorm:"primaryKey" json:"id"`
	DoctorID    uuid.UUID  `gorm:"not null;index" json:"doctor_id"`
	Action      string     `gorm:"size:100;not null" json:"action"`
	Status      string     `gorm:"size:50;not null" json:"status"`
	Remarks     *string    `gorm:"type:text" json:"remarks,omitempty"`
	PerformedBy *uuid.UUID `json:"performed_by,omitempty"`
	PerformedAt time.Time  `gorm:"autoCreateTime" json:"performed_at"`
}

func (h *VerificationHistory) BeforeCreate(tx *gorm.DB) (err error) {
	if h.ID == uuid.Nil {
		h.ID = uuid.New()
	}
	return nil
}

File: backend_api/entities/history.py
import uuid
from sqlalchemy import Column, String, DateTime, Text, ForeignKey
from sqlalchemy.sql import func
from database import Base

class VerificationHistory(Base):
    __tablename__ = "verification_histories"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    doctor_id = Column(String(36), ForeignKey("doctors.id", ondelete="CASCADE"), nullable=False, index=True)
    action = Column(String(100), nullable=False)
    status = Column(String(50), nullable=False)
    remarks = Column(Text, nullable=True)
    performed_by = Column(String(36), nullable=True)
    performed_at = Column(DateTime(timezone=True), server_default=func.now())
